crates/libs/core/src/interface.rs

```diff
@@ -1,11 +1,15 @@
+use super::*;
+
 /// Provides low-level access to an interface vtable.
 ///
 /// This trait is automatically implemented by the generated bindings and should not be
 /// implemented manually.
 ///
 /// # Safety
-pub unsafe trait Interface: Sized {
+pub unsafe trait Interface: Sized + Clone {
     type Vtable;
+    const IID: GUID;
+    const UNKNOWN: bool = true;
```

Comment:
Would it make sense to combine these into an `Option<GUID>`?

Reply:
It could be made to work, but it would again complicate the nominal case of COM interfaces.

Reply:
Sorry, didn't knew that. I was always under the impression that both (deriving from `IUnknown` and having a unique IID) would be required for a COM interface. 

I hope you don't mind me asking, but do you happen to know if such interfaces exist in the Windows API? I am curious in which situations this would be useful.

Reply:
Sure, there are probably a few dozen in the Windows SDK. They're generally used as callback interfaces where some caller may provide the implementation, but it doesn't necessarily represent an interface on a COM object that you'd query for.

Reply:
Just to make sure I'm not misunderstanding, do any of these actually implement `IUnknown`?
I know about `IXAudio2VoiceCallback` or `ISpNotifyCallback`, for example, but these do not inherit from `IUnknown` (and are therefore not considered to be COM interfaces (?)).

Also, what would be the reason for such an callback interface to implement `IUnknown` if you're not supposed to query for it? Just the `AddRef`/`Release` ref-counting functionality?

Reply:
Yes, lifetime management.